django_celery_results/views.py

```diff
@@ -44,8 +44,12 @@ def is_group_successful(request, group_id):
 
 
 def group_status(request, group_id):
-    """Return task status and result in JSON format."""
+    """Return group id and its async results status & result in JSON format."""
     result = GroupResult.restore(group_id)
-    retval = result.result
-    response_data = {'id': group_id, 'result': retval}
-    return JsonResponse({'task': response_data})
+    retval = [
+        {"result": async_result.result, "status": async_result.status}
+        for async_result in result.results
+    ]
```

Comment:
Consider using `GroupResult.as_tuple()` to also return nested groups.

This has the added benefit of mainly using the GroupResult API to serialise the GroupResult instead of reimplementing it.

Reply:
hmm. It looks like `as_tuple` doesn't include the status of the tasks, so im not sure we'll be able to use it here. I hadn't thought abut nested groups though, good catch.

Reply:
Ah right it just contains the IDs